packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -270,10 +328,9 @@ export class BasicCrawler<
     autoscaledPool?: AutoscaledPool;
 
     protected log: Log;
-    protected userProvidedHandler: HandleRequest<Context>;
-    protected failedContextHandler?: HandleFailedRequest<ErrorContext>;
-    protected handleFailedRequestFunction?: HandleFailedRequest<ErrorContext>;
-    protected handleRequestTimeoutMillis: number;
+    protected userProvidedHandler!: RequestHandler<Context>;
```

Comment:
lets rename this, the handler is always "user provided", i'd use just `requestHandler`. also let's inline `_handleRequestHandler`.

btw i thought this will be much easier, so many changes :P

Reply:
Inlining it will be fun, since the method gets overridden in classes that extend it. Should be doable

Reply:
if it makes sense to keep it, lets keep it, i just saw it here without any purpose. but the name is quite meh, "handle handler" :]

Reply:
I'd say let's keep it (since extensions of basic crawler can augment the context), and I agree, the naming could be improved.. runRequestHandler maybe?